services/spar/test-integration/Test/Spar/APISpec.hs

```diff
@@ -618,14 +618,14 @@ specCRUDIdentityProvider = do
           callIdpGet' (env ^. teSpar) (Just userid) idpid
             `shouldRespondWith` checkErr (== 404) "not-found"
 
-        context "with email" $ it "responds with 2xx and removes IdP" $ do
+        context "with email, idp non-empty" $ it "responds with 412 and does not remove IdP" $ do
           env <- ask
           (firstOwner, tid, idp) <- registerTestIdP
           ssoOwner <- mkSsoOwner firstOwner tid idp
           callIdpDelete' (env ^. teSpar) (Just ssoOwner) (idp ^. idpId)
-            `shouldRespondWith` \resp -> statusCode resp < 300
+            `shouldRespondWith` checkErr (== 412) "idp-has-bound-users"
           callIdpGet' (env ^. teSpar) (Just ssoOwner) (idp ^. idpId)
-            `shouldRespondWith` checkErr (== 404) "not-found"
+            `shouldRespondWith` \resp -> statusCode resp < 300
```

Comment:
I don't have an explanation for this change, though.  This is indeed odd.

Reply:
ah, no, should have read the test: if spar refuses to delete the idp, it won't be deleted, so getting it will still work.  :)